src/interpreter.coffee

```diff
@@ -0,0 +1,118 @@
+_ = window?._ ? self?._ ? global?._ ? require 'lodash'
+
+
+updateState = (aether, evaluator) ->
+  frame_stack = evaluator.frames
+  top = frame_stack[0]
+  bottom = frame_stack[frame_stack.length - 1]
+
+  if aether.options.includeFlow
+    unless bottom.flow?
+      bottom.flow = {statementsExecuted: 0, statements: []}
+      aether.flow.states ?= []
+      aether.flow.states.push bottom.flow
+
+  if aether.options.includeMetrics
+    aether.metrics.statementsExecuted ?= 0
+    aether.metrics.callsExecuted ?= 0
+  
+  astStack = (x.ast for x in frame_stack when x.ast?)
+
+  if astStack[0]? 
+    rng = astStack[0].originalRange
+    aether.lastStatementRange = [rng.start, rng.end] if rng
+
+  if top.ast?
+    ++aether.metrics.callsExecuted if aether.options.includeMetrics and top.ast.type == 'CallExpression'
+
+    unless top.ast.type in ['Literal', 'Identifier', 'ThisExpression', 'BlockStatement', 'MemberExpression', 'FunctionExpression']
```

Comment:
Could golden path this and the `if bottom.flow?` check with early returns to save on nesting and make the default execution case more clear.